File: docs/_static/codes/get_started/coppeliasim_basics/communication/communication_test.py
#!/bin/python3
from dqrobotics.interfaces.vrep import DQ_VrepInterface
import time




def main() -> None:
    vi = DQ_VrepInterface()
    try:
        if not vi.connect("127.0.0.1", 19997, 100, 10):
            raise RuntimeError("Unable to connect to CoppeliaSim.")

        vi.start_simulation()
        time.sleep(0.1)
        x = vi.get_object_pose("/Frame_x")
        print("Position: ", x.translation())
        print("The test was successful!")

    except (Exception, KeyboardInterrupt) as e:
        print(e)
        pass

    finally:
        vi.stop_simulation()
        vi.disconnect()


if __name__ == "__main__":
    main()

File: docs/_static/codes/get_started/coppeliasim_basics/communication/communication_test.cpp
#include <iostream>
#include <dqrobotics/interfaces/vrep/DQ_VrepInterface.h>
#include <thread>

int main()
{
    auto vi = DQ_VrepInterface();
    try {
        vi.connect("10.198.113.136", 19997,100,10);
        vi.start_simulation();
        std::this_thread::sleep_for(std::chrono::milliseconds(100));
        DQ x = vi.get_object_pose("Frame_x");
        vi.stop_simulation();
        vi.disconnect();
        std::cout<<"Position: "<<x.translation()<<std::endl;
        std::cout<<"the test was successful!"<<std::endl;
    } catch (std::exception& e) {
        std::cout<<e.what()<<std::endl;
        vi.stop_simulation();
        vi.disconnect();
    }
    return 0;
}